File: examples/python/research_agent.py
import sys
import os

# Add local path for testing
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '../../bindings/python')))

from openweave import Agent

def main():
    agent = Agent(system_prompt="You are a research agent.")
    print("Running research agent...")
    res = agent.run("Research the history of Rust programming language")
    print(res)

if __name__ == "__main__":
    main()

File: examples/javascript/research_agent.ts
import { Agent } from '../../bindings/javascript/dist';

async function main() {
    const agent = new Agent({
        system_prompt: "You are a research agent."
    });
    
    console.log("Running research agent...");
    try {
        const res = await agent.run("Research the history of Rust programming language");
        console.log(res);
    } catch (e) {
        console.error(e);
    } finally {
        agent.destroy();
    }
}

main();